components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1697,10 +1734,29 @@ public void deleteGroup(String groupId) throws NotFoundException, CharonExceptio
     @Override
     public List<Object> listGroupsWithGET(Node rootNode, int startIndex, int count, String sortBy, String sortOrder,
             String domainName, Map<String, Boolean> requiredAttributes)
+
             throws CharonException, NotImplementedException, BadRequestException {
         if (sortBy != null || sortOrder != null) {
             throw new NotImplementedException("Sorting is not supported");
-        } else if (startIndex != 1) {
+        } else if (startIndex != 1 && count >= 0) {
+            throw new NotImplementedException("Pagination is not supported");
+        } else if (rootNode != null) {
+            return filterGroups(rootNode, requiredAttributes);
+        } else {
+            return listGroups(requiredAttributes);
+        }
+    }
+
+    @Override
+    public List<Object> listGroupsWithGET(Node rootNode, Integer startIndex, Integer count, String sortBy,
+            String sortOrder, String domainName, Map<String, Boolean> requiredAttributes)
+            throws CharonException, NotImplementedException {
+
+        // Validate NULL value for startIndex.
+        startIndex = handleStartIndexEqualsNULL(startIndex);
+        if (sortBy != null || sortOrder != null) {
+            throw new NotImplementedException("Sorting is not supported");
+        } else if (startIndex != 1 || count != null) {
```

Comment:
Do we need to handle null start index above.
Simple it's not null means and in input is provided right.
So just mentioning it's not supported if it's not null is enough right

Reply:
Meanwhile it's consistent with other arguments being used here and more readable

Reply:
according to the spec, start Index param is optional. In that case we have to set it to 1 as it is in the specification. So mentioning as "Not Supported" is not an option.

Reply:
But we are not implementing pagination support for `listGroupsWithGET` right? Then For this implementation what needs to be returned is pagination is not supported, if user request comes with a start index. Even though you handle startIndex correctly, it won't be used unless pagination is properly implemented.

Reply:
Refer to line 1758. If the startIndex value is not equal to 1, it throws an exception. The Null implementation for startIndex is correct because returning 1 is the default case.

Reply:
Ok, I see the point. This way both `null` and `1` cases are handled, and anything else means the pagination which we don't support.

Reply:
+1